### ISSUE TITLE
Should we move type annotations to stub files?

### ISSUE THREAD

**Issue by jindrahelcl** [ufal/neuralmonkey#98]
According to [PEP 484](https://www.python.org/dev/peps/pep-0484/#stub-files), type annotations of methods can be placed in separate files.

Pros:
- Use of Python 3.4 will (likely) be again an option. At MT Marathon, the version of Python was among the most frequent problems.

Cons:
- The type annotations won't be placed (for some of us) conveniently in the code.

I am spíš pro..

**Comment by jlibovicky** [comment]
Why is exactly if Python 3.4 easier to use than 3.5? And is there some outlook, like in few months everyone will be using 3.5 or will it be still an issue? How many user will bring to us? (Indeed, I am lazy to do anything, that's why I'am asking these questions.)

**Comment by jindrahelcl** [comment]
I think it's mostly a temporary issue with the Python version.
I just remembered another advantage of separated annotations - readthedocs cannot deal with them (because they use 3.4) so the documentation for these files cannot be rendered.

While opening this issue, I did not intend to bring the 3.5 vs 3.4 question - instead, I wanted to open discussion about the type annotation style  regardless of the Python version. For example, @hajicj hates the 3.5 type hints and this could help him see them in a better light. :-)
